ipa-core/src/protocol/ipa_prf/boolean_ops/addition_sequential.rs

```diff
@@ -258,9 +283,32 @@ mod test {
                     .unwrap()
                 })
                 .await
-                .reconstruct()
-                .as_u128();
-            assert_eq!((x, y, result), (x, y, expected));
+                .reconstruct();
+            assert_eq!(
+                (x, y, result.as_u128(), carry.as_u128()),
+                (x, y, expected, expected_carry)
+            );
+
+            let x = x & ((1 << 32) - 1);
```

Comment:
Isn't this a second test?

Reply:
It's still testing addition of different lengths, just in the opposite order.